Shadowserver 202207

elsif2 opened the pull request:
Large set of changes previously submitted as 2143.

elsif2 committed:
Replaced misleading _country_ parameter with _reports_.

elsif2 committed:
parent ba7d24bd79abf77ddcf92198f5150e2b0fffa383
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659452139 +0000

parent ba7d24bd79abf77ddcf92198f5150e2b0fffa383
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659452135 +0000

parent ba7d24bd79abf77ddcf92198f5150e2b0fffa383
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659452132 +0000

parent ba7d24bd79abf77ddcf92198f5150e2b0fffa383
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659452116 +0000

parent ba7d24bd79abf77ddcf92198f5150e2b0fffa383
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659452111 +0000

parent ba7d24bd79abf77ddcf92198f5150e2b0fffa383
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659452106 +0000

parent ba7d24bd79abf77ddcf92198f5150e2b0fffa383
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659451910 +0000

parent ba7d24bd79abf77ddcf92198f5150e2b0fffa383
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659451905 +0000

parent ba7d24bd79abf77ddcf92198f5150e2b0fffa383
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659451900 +0000

parent ba7d24bd79abf77ddcf92198f5150e2b0fffa383
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659451775 +0000

Update parser to support all available reports.

Update to existing test cases to match current report types.

New tests for added report types.

pycodestyle fixes

add testdata licenses

pycodestyle fix

Added reports parameter

Suggested changes to the parser

Proposed details for the release

Test script updates for suggested changes

Test input updates

Realign columns

Update compromised_website.csv

Update scan_adb.csv

Update scan_adb.csv

Update scan_ftp.csv

Update scan_ipp.csv

Update scan_snmp.csv

Realign columns

Remove duplicates

Changed malware.name to extra.infection

Updated SPDX-FileCopyrightText

shadowserver api: document and warn on old parameter

document the old parameter `country` and its status
warn if used
adapt the test

DOC: fix NEWS entry of PR#2143

Added the sector field to scan_amqp, scan_cwmp, and scan_vnc.

Copyright and raw field updates

Added the sector field to scan_amqp, scan_cwmp, and scan_vnc.

Copyright updates

Added phish_url and scan_modbus reports.

Update source.url and source.fqdn for phish_url and malware_url reports.  Update classification.taxonomy and classification.type for scan_modbus report.

* additional field type validation changes
* added count, bytes, duration, avg_pps, and max_pps fields to event_honeypot_ddos_amp
* added 'protocol.application': 'https' to scan_ssl, scan_ssl_freak, and scan_ssl_poodle
* added 'extra.tag' to scan_* and device_id

Replaced scan_modbus with scan_ics

Addeed event4_honeypot_ddos, event4_honeypot_ddos_target, scan_dvr_dhcpdiscover, and scan_socks.

Tests for event4_honeypot_ddos.

Tests for event4_honeypot_ddos_target.

Tests for scan_dvr_dhcpdiscover.

Tests for scan_socks.

Rename file

Rename file

update:scan_mdns, scan_smb, and special; add:scan_ddos_middle_box

cleanup renamed license files

updated scan_mdns test files

updated scan_smb test files

updated special test files

add scan_ddos_middlebox test files

add scan_ddos_middlebox test

updated schema

Updated scan_smb tests

Updated scan_ntp tests

Updated scan_snmp tests

New scan_docker test

New scan_kubernetes test

New scan_mysql test

Updated report schema for June 2022

Added scan_epmd test

Revert "Added scan_epmd test"

This reverts commit 01edea18d82b583290f66eae5f2457f4695623cd.

Revert: Fix for recover_line method as commited in certtools#2192

Added scan_couchdb

Test case for scan_couchdb

Added scan6_rpd

Added/updated README with maintainer details

Restored feed names and classification.identifiers to minimize upgrade impact.

Merge repair

pycodestyle repairs

codespell fixes

license compliance fixes

pycodestyle fixes

Feed configuration updates for compatibility with the original.

Added scan_postgres test

Added additional IPv6 aliases

Fix for recover_line method as commited in certtools#2192

parent 61940147ef9c5804f4168dcd57a57cfd12053328
author elsif2 <elsif@shadowserver.org> 1659389509 +0000
committer elsif2 <elsif@shadowserver.org> 1659392927 +0000

parent 61940147ef9c5804f4168dcd57a57cfd12053328
author elsif2 <elsif@shadowserver.org> 1659389509 +0000
committer elsif2 <elsif@shadowserver.org> 1659392915 +0000

parent 61940147ef9c5804f4168dcd57a57cfd12053328
author elsif2 <elsif@shadowserver.org> 1659389509 +0000
committer elsif2 <elsif@shadowserver.org> 1659392880 +0000

parent 61940147ef9c5804f4168dcd57a57cfd12053328
author elsif2 <elsif@shadowserver.org> 1659389509 +0000
committer elsif2 <elsif@shadowserver.org> 1659392820 +0000

parent 61940147ef9c5804f4168dcd57a57cfd12053328
author elsif2 <elsif@shadowserver.org> 1659389509 +0000
committer elsif2 <elsif@shadowserver.org> 1659392766 +0000

parent 61940147ef9c5804f4168dcd57a57cfd12053328
author elsif2 <elsif@shadowserver.org> 1659389509 +0000
committer elsif2 <elsif@shadowserver.org> 1659392468 +0000

parent 61940147ef9c5804f4168dcd57a57cfd12053328
author elsif2 <elsif@shadowserver.org> 1659389509 +0000
committer elsif2 <elsif@shadowserver.org> 1659392457 +0000

parent 61940147ef9c5804f4168dcd57a57cfd12053328
author elsif2 <elsif@shadowserver.org> 1659389509 +0000
committer elsif2 <elsif@shadowserver.org> 1659392039 +0000

Update parser to support all available reports.

Update to existing test cases to match current report types.

New tests for added report types.

pycodestyle fixes

add testdata licenses

pycodestyle fix

Added reports parameter

Suggested changes to the parser

Proposed details for the release

Test script updates for suggested changes

Test input updates

Realign columns

Update compromised_website.csv

Update scan_adb.csv

Update scan_adb.csv

Update scan_ftp.csv

Update scan_ipp.csv

Update scan_snmp.csv

Realign columns

Remove duplicates

Changed malware.name to extra.infection

Updated SPDX-FileCopyrightText

shadowserver api: document and warn on old parameter

document the old parameter `country` and its status
warn if used
adapt the test

DOC: fix NEWS entry of PR#2143

Added the sector field to scan_amqp, scan_cwmp, and scan_vnc.

Copyright and raw field updates

Added the sector field to scan_amqp, scan_cwmp, and scan_vnc.

Copyright updates

Added phish_url and scan_modbus reports.

Update source.url and source.fqdn for phish_url and malware_url reports.  Update classification.taxonomy and classification.type for scan_modbus report.

* additional field type validation changes
* added count, bytes, duration, avg_pps, and max_pps fields to event_honeypot_ddos_amp
* added 'protocol.application': 'https' to scan_ssl, scan_ssl_freak, and scan_ssl_poodle
* added 'extra.tag' to scan_* and device_id

Replaced scan_modbus with scan_ics

Addeed event4_honeypot_ddos, event4_honeypot_ddos_target, scan_dvr_dhcpdiscover, and scan_socks.

Tests for event4_honeypot_ddos.

Tests for event4_honeypot_ddos_target.

Tests for scan_dvr_dhcpdiscover.

Tests for scan_socks.

Rename file

Rename file

update:scan_mdns, scan_smb, and special; add:scan_ddos_middle_box

cleanup renamed license files

updated scan_mdns test files

updated scan_smb test files

updated special test files

add scan_ddos_middlebox test files

add scan_ddos_middlebox test

updated schema

Updated scan_smb tests

Updated scan_ntp tests

Updated scan_snmp tests

New scan_docker test

New scan_kubernetes test

New scan_mysql test

Updated report schema for June 2022

Added scan_epmd test

Revert "Added scan_epmd test"

This reverts commit 01edea18d82b583290f66eae5f2457f4695623cd.

Revert: Fix for recover_line method as commited in certtools#2192

Added scan_couchdb

Test case for scan_couchdb

Added scan6_rpd

Added/updated README with maintainer details

Restored feed names and classification.identifiers to minimize upgrade impact.

Merge repair

pycodestyle repairs

codespell fixes

license compliance fixes

pycodestyle fixes

Feed configuration updates for compatibility with the original.

Added scan_postgres test

Added additional IPv6 aliases

Fix for recover_line method as commited in certtools#2192

Replaced misleading _country_ parameter with _reports_.

Update parser to support all available reports.

Update to existing test cases to match current report types.

pycodestyle fixes

add testdata licenses

pycodestyle fix

Suggested changes to the parser

Proposed details for the release

Test script updates for suggested changes

Test input updates

Realign columns

Update compromised_website.csv

Update scan_adb.csv

Update scan_ftp.csv

Update scan_ipp.csv

Realign columns

Remove duplicates

Changed malware.name to extra.infection

DOC: fix NEWS entry of PR#2143

Added the sector field to scan_amqp, scan_cwmp, and scan_vnc.

Copyright updates

Added phish_url and scan_modbus reports.

Update source.url and source.fqdn for phish_url and malware_url reports.  Update classification.taxonomy and classification.type for scan_modbus report.

* additional field type validation changes
* added count, bytes, duration, avg_pps, and max_pps fields to event_honeypot_ddos_amp
* added 'protocol.application': 'https' to scan_ssl, scan_ssl_freak, and scan_ssl_poodle
* added 'extra.tag' to scan_* and device_id

Replaced scan_modbus with scan_ics

Tests for event4_honeypot_ddos.

Tests for event4_honeypot_ddos_target.

Tests for scan_dvr_dhcpdiscover.

Tests for scan_socks.

Rename file

Rename file

cleanup renamed license files

updated scan_mdns test files

updated scan_smb test files

updated special test files

add scan_ddos_middlebox test files

add scan_ddos_middlebox test

Updated scan_ntp tests

Updated scan_snmp tests

New scan_docker test

New scan_kubernetes test

New scan_mysql test

Added scan_epmd test

Revert "Added scan_epmd test"

This reverts commit 01edea18d82b583290f66eae5f2457f4695623cd.

Fix for recover_line method as commited in #2192

Revert: Fix for recover_line method as commited in certtools#2192

Test case for scan_couchdb

Restored feed names and classification.identifiers to minimize upgrade impact.

Added function to notify if any unsupported feeds are configured.

pycodestyle repairs

codespell fixes

license compliance fixes

pycodestyle fixes

Feed configuration updates for compatibility with the original.

Added scan_postgres test

Update intelmq/bots/collectors/shadowserver/collector_reports_api.py

Co-authored-by: Sebastian <sebix@sebix.at>

Update intelmq/bots/collectors/shadowserver/README.md

Co-authored-by: Sebastian <sebix@sebix.at>

Update intelmq/tests/bots/parsers/shadowserver/README.md

Co-authored-by: Sebastian <sebix@sebix.at>

resync to develop

parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659395284 +0000

parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659395281 +0000

parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659395278 +0000

parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659395264 +0000

parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659395260 +0000

parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659395256 +0000

parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659395141 +0000

parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659395131 +0000

parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659395127 +0000

parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659395122 +0000

parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
author elsif2 <elsif@shadowserver.org> 1643216571 +0000
committer elsif2 <elsif@shadowserver.org> 1659395058 +0000

Update parser to support all available reports.

Update to existing test cases to match current report types.

New tests for added report types.

pycodestyle fixes

add testdata licenses

pycodestyle fix

Added reports parameter

Suggested changes to the parser

Proposed details for the release

Test script updates for suggested changes

Test input updates

Realign columns

Update compromised_website.csv

Update scan_adb.csv

Update scan_adb.csv

Update scan_ftp.csv

Update scan_ipp.csv

Update scan_snmp.csv

Realign columns

Remove duplicates

Changed malware.name to extra.infection

Updated SPDX-FileCopyrightText

shadowserver api: document and warn on old parameter

document the old parameter `country` and its status
warn if used
adapt the test

DOC: fix NEWS entry of PR#2143

Added the sector field to scan_amqp, scan_cwmp, and scan_vnc.

Copyright and raw field updates

Added the sector field to scan_amqp, scan_cwmp, and scan_vnc.

Copyright updates

Added phish_url and scan_modbus reports.

Update source.url and source.fqdn for phish_url and malware_url reports.  Update classification.taxonomy and classification.type for scan_modbus report.

* additional field type validation changes
* added count, bytes, duration, avg_pps, and max_pps fields to event_honeypot_ddos_amp
* added 'protocol.application': 'https' to scan_ssl, scan_ssl_freak, and scan_ssl_poodle
* added 'extra.tag' to scan_* and device_id

Replaced scan_modbus with scan_ics

Addeed event4_honeypot_ddos, event4_honeypot_ddos_target, scan_dvr_dhcpdiscover, and scan_socks.

Tests for event4_honeypot_ddos.

Tests for event4_honeypot_ddos_target.

Tests for scan_dvr_dhcpdiscover.

Tests for scan_socks.

Rename file

Rename file

update:scan_mdns, scan_smb, and special; add:scan_ddos_middle_box

cleanup renamed license files

updated scan_mdns test files

updated scan_smb test files

updated special test files

add scan_ddos_middlebox test files

add scan_ddos_middlebox test

updated schema

Updated scan_smb tests

Updated scan_ntp tests

Updated scan_snmp tests

New scan_docker test

New scan_kubernetes test

New scan_mysql test

Updated report schema for June 2022

Added scan_epmd test

Revert "Added scan_epmd test"

This reverts commit 01edea18d82b583290f66eae5f2457f4695623cd.

Revert: Fix for recover_line method as commited in certtools#2192

Added scan_couchdb

Test case for scan_couchdb

Added scan6_rpd

Added/updated README with maintainer details

Restored feed names and classification.identifiers to minimize upgrade impact.

Merge repair

pycodestyle repairs

codespell fixes

license compliance fixes

pycodestyle fixes

Feed configuration updates for compatibility with the original.

Added scan_postgres test

Added additional IPv6 aliases

Fix for recover_line method as commited in certtools#2192

resolve conflict

Replaced misleading _country_ parameter with _reports_.

Update parser to support all available reports.

Update to existing test cases to match current report types.

New tests for added report types.

pycodestyle fixes

add testdata licenses

pycodestyle fix

Added reports parameter

Suggested changes to the parser

Proposed details for the release

Test script updates for suggested changes

Test input updates

Realign columns

Update compromised_website.csv

Update scan_adb.csv

Update scan_adb.csv

Update scan_ftp.csv

Update scan_ipp.csv

Update scan_snmp.csv

Realign columns

Remove duplicates

Changed malware.name to extra.infection

Updated SPDX-FileCopyrightText

shadowserver api: document and warn on old parameter

document the old parameter `country` and its status
warn if used
adapt the test

DOC: fix NEWS entry of PR#2143

Added the sector field to scan_amqp, scan_cwmp, and scan_vnc.

Copyright and raw field updates

Added the sector field to scan_amqp, scan_cwmp, and scan_vnc.

Copyright updates

Added phish_url and scan_modbus reports.

Update source.url and source.fqdn for phish_url and malware_url reports.  Update classification.taxonomy and classification.type for scan_modbus report.

* additional field type validation changes
* added count, bytes, duration, avg_pps, and max_pps fields to event_honeypot_ddos_amp
* added 'protocol.application': 'https' to scan_ssl, scan_ssl_freak, and scan_ssl_poodle
* added 'extra.tag' to scan_* and device_id

Replaced scan_modbus with scan_ics

Addeed event4_honeypot_ddos, event4_honeypot_ddos_target, scan_dvr_dhcpdiscover, and scan_socks.

Tests for event4_honeypot_ddos.

Tests for event4_honeypot_ddos_target.

Tests for scan_dvr_dhcpdiscover.

Tests for scan_socks.

Rename file

Rename file

update:scan_mdns, scan_smb, and special; add:scan_ddos_middle_box

cleanup renamed license files

updated scan_mdns test files

updated scan_smb test files

updated special test files

add scan_ddos_middlebox test files

add scan_ddos_middlebox test

updated schema

Updated scan_smb tests

Updated scan_ntp tests

Updated scan_snmp tests

New scan_docker test

New scan_kubernetes test

New scan_mysql test

Updated report schema for June 2022

Added scan_epmd test

Revert "Added scan_epmd test"

This reverts commit 01edea18d82b583290f66eae5f2457f4695623cd.

Revert: Fix for recover_line method as commited in certtools#2192

Added scan_couchdb

Test case for scan_couchdb

Added scan6_rpd

Added/updated README with maintainer details

Restored feed names and classification.identifiers to minimize upgrade impact.

Merge repair

pycodestyle repairs

codespell fixes

license compliance fixes

pycodestyle fixes

Feed configuration updates for compatibility with the original.

Added scan_postgres test

Added additional IPv6 aliases

Fix for recover_line method as commited in certtools#2192

Update to existing test cases to match current report types.

add testdata licenses

pycodestyle fix

Proposed details for the release

Test script updates for suggested changes

Realign columns

Update compromised_website.csv

Update scan_adb.csv

Update scan_ftp.csv

Update scan_ipp.csv

Realign columns

Remove duplicates

DOC: fix NEWS entry of PR#2143

Added the sector field to scan_amqp, scan_cwmp, and scan_vnc.

Copyright updates

Added phish_url and scan_modbus reports.

Update source.url and source.fqdn for phish_url and malware_url reports.  Update classification.taxonomy and classification.type for scan_modbus report.

* additional field type validation changes
* added count, bytes, duration, avg_pps, and max_pps fields to event_honeypot_ddos_amp
* added 'protocol.application': 'https' to scan_ssl, scan_ssl_freak, and scan_ssl_poodle
* added 'extra.tag' to scan_* and device_id

Replaced scan_modbus with scan_ics

Tests for event4_honeypot_ddos.

Tests for event4_honeypot_ddos_target.

Tests for scan_dvr_dhcpdiscover.

Tests for scan_socks.

Rename file

Rename file

cleanup renamed license files

updated scan_mdns test files

updated special test files

add scan_ddos_middlebox test files

add scan_ddos_middlebox test

New scan_docker test

New scan_kubernetes test

New scan_mysql test

Added scan_epmd test

Revert "Added scan_epmd test"

This reverts commit 01edea18d82b583290f66eae5f2457f4695623cd.

Fix for recover_line method as commited in #2192

Revert: Fix for recover_line method as commited in certtools#2192

Test case for scan_couchdb

Restored feed names and classification.identifiers to minimize upgrade impact.

Added function to notify if any unsupported feeds are configured.

pycodestyle repairs

codespell fixes

license compliance fixes

pycodestyle fixes

Feed configuration updates for compatibility with the original.

Added scan_postgres test

Update intelmq/bots/collectors/shadowserver/collector_reports_api.py

Co-authored-by: Sebastian <sebix@sebix.at>

Update intelmq/bots/collectors/shadowserver/README.md

Co-authored-by: Sebastian <sebix@sebix.at>

Update intelmq/tests/bots/parsers/shadowserver/README.md

Co-authored-by: Sebastian <sebix@sebix.at>

resync to develop

sebix commented on docs/user/bots.rst:
```diff
@@ -654,9 +654,10 @@ The Cache is required to memorize which files have already been processed (TTL n
+* `country`: **Deprecated:** The country you want to download the reports for. Will be removed in IntelMQ version 4.0.0, use *reports* instead.
 * `apikey`: Your Shadowserver API key
 * `secret`: Your Shadowserver API secret
+* `reports`: A list of strings or a comma-separated list of the mailing lists you want to process.
```
What is a "mailing list" in this context?

elsif2 replied:
Mailing lists are how reports were delivered before API delivery was available.  Some organizations had multiple mailing lists based on a filter type (e.g. geo and IP/CIDR).

sebix commented on intelmq/bots/collectors/shadowserver/collector_reports_api.py:
```diff
@@ -22,12 +23,14 @@
         api_key (str): Your Shadowserver API key
         secret (str): Your Shadowserver API secret
-        country (str): The country you want to download reports for (i.e. 'austria')
+        country (str): DEPRECIATED The mailing list you want to download reports for (i.e. 'austria')
```
```suggestion
        country (str): DEPRECATED The mailing list you want to download reports for (i.e. 'austria')
```

sebix commented on intelmq/bots/collectors/shadowserver/README.md:
```diff
@@ -0,0 +1,9 @@
+This module is maintained by [The Shadowserver Foundation](https://www.shadowserver.org/).  
+
+Please contact intelmq@shadowserver.org with any issues or concerns.
+
```
```suggestion
```

sebix commented on intelmq/bots/parsers/shadowserver/README.md:
```diff
@@ -0,0 +1,9 @@
+This module is maintained by [The Shadowserver Foundation](https://www.shadowserver.org/).  
+
+Please contact intelmq@shadowserver.org with any issues or concerns.
+
```
```suggestion
```

sebix commented on intelmq/lib/test.py:
```diff
@@ -46,7 +46,7 @@
 
 
-class Parameters:
+class Parameters(object):
```
```suggestion
class Parameters():
```
Adding object as parent is not necessary on Python 3 any more

elsif2 replied:
I don't believe I modified intelmq/lib/test.py - this must be a synchronization issue

sebix commented on intelmq/lib/test.py:
```diff
@@ -63,7 +63,7 @@ def mocked(conf_file):
             fname = pkg_resources.resource_filename('intelmq',
                                                     confname)
-            with open(fname) as fpconfig:
+            with open(fname, 'rt') as fpconfig:
```
why is this change needed here?

sebix commented on intelmq/lib/test.py:
```diff
@@ -105,7 +105,7 @@ def skip_build_environment():
 
 
-class BotTestCase:
+class BotTestCase(object):
```
```suggestion
class BotTestCase():
```

sebix commented on intelmq/lib/test.py:
```diff
@@ -199,11 +199,12 @@ def prepare_bot(self, parameters={}, destination_queues=None):
-        src_name = f"{self.bot_id}-input"
+        src_name = "{}-input".format(self.bot_id)
         if not destination_queues:
-            destination_queues = {"_default": f"{self.bot_id}-output"}
+            destination_queues = {"_default": "{}-output".format(self.bot_id)}
         else:
-            destination_queues = {queue_name: f"{self.bot_id}-{queue_name.strip('_')}-output"
+            destination_queues = {queue_name: "%s-%s-output" % (self.bot_id,
+                                                                queue_name.strip('_'))
```
I don't see a relation to the shadowserver changes

sebix commented on intelmq/tests/bots/parsers/shadowserver/README.md:
```diff
@@ -14,3 +15,8 @@ each type of report, there are also the following tests:
+This module is maintained by [The Shadowserver Foundation](https://www.shadowserver.org/).  
+
+Please contact intelmq@shadowserver.org with any issues or concerns.
+
```
```suggestion
```

sebix reviewed (commented):
Would appreciate if you could squelch the commits together (not necessarily to one, but < 10)

elsif2 commented:
I squashed the changes into a small number of commits.

sebix commented on CHANGELOG.md:
```diff
@@ -100,6 +100,8 @@ CHANGELOG
 - `intelmq.bots.outputs.templated_smtp.output`:
   - Add new function `from_json()` (which just calls `json.loads()` in the standard Python environment), meaning the Templated SMTP output bot can take strings containing JSON documents and do the formatting itself (PR#2120 by Karl-Johan Karlsson).
   - Lift restriction on requirement jinja2 < 3 (PR#2158 by Sebastian Wagner).
+- `intelmq.bots.outputs.sql`:
+  - For PostgreSQL, escape Nullbytes in text to prevent "unsupported Unicode escape sequence" issues (PR#2223 by Sebastian Wagner, fixes #2203).
```
This change shouldn't be here

sebix commented on intelmq/lib/test.py:
```diff
@@ -188,7 +188,7 @@ def log(name, *args, **kwargs):
         return log
 
-    def prepare_bot(self, parameters={}, destination_queues=None):
+    def prepare_bot(self, parameters={}, destination_queues=None, prepare_source_queue: bool = True):
```
There are also still changes in unrelated parts of the code

elsif2 replied:
I don't see any changes compared to the develop branch:

```
$ git status
On branch shadowserver-202207
Your branch is up to date with 'origin/shadowserver-202207'.

nothing to commit, working tree clean
```

```
$ curl -s 'https://raw.githubusercontent.com/certtools/intelmq/develop/intelmq/lib/test.py' > /tmp/test.py
$ diff -u intelmq/lib/test.py /tmp/test.py
$
```

sebix commented:
> I squashed the changes into a small number of commits.

This PR has 95 commits?

elsif2 commented:
> > I squashed the changes into a small number of commits.
> This PR has 95 commits?

I did a `git rebase -i develop` and squashed down to three which required a lot of merge repairs.

The git log shows them as follows:

```
commit 67a52fe8e8d4c965970af8884ae627e272d9002c
Author: elsif2 <elsif@shadowserver.org>
Date:   Wed Jan 26 17:02:51 2022 +0000

    parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
    author elsif2 <elsif@shadowserver.org> 1643216571 +0000
    committer elsif2 <elsif@shadowserver.org> 1659395284 +0000
    
    parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
    author elsif2 <elsif@shadowserver.org> 1643216571 +0000
    committer elsif2 <elsif@shadowserver.org> 1659395281 +0000
    
    parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
    author elsif2 <elsif@shadowserver.org> 1643216571 +0000
    committer elsif2 <elsif@shadowserver.org> 1659395278 +0000
    
    parent 7dc5b7471b952d4fd703c2d8921ea992296194f3
    author elsif2 <elsif@shadowserver.org> 1643216571 +0000
    committer elsif2 <elsif@shadowserver.org> 1659395264 +0000
 .
 .
 .
```

sebix commented:
GitHub says 95 commits
![image](https://user-images.githubusercontent.com/199050/182397424-55e35e55-19c8-49e3-ad72-db8f01fb3536.png)
command line says 93:
```
$ git log --oneline  upstream/pr/2227...develop | wc -l
93
```

elsif2 commented:
What would you suggest?

sebix commented:
```
git rebase -i develop
```
then squash various commit together and
```
git push -f origin shadowserver-202207
```

elsif2 committed:
merge repair

elsif2 commented:
> ```
> git rebase -i develop
> ```
> 
> then squash various commit together and
> 
> ```
> git push -f origin shadowserver-202207
> ```

Done:

> [elsif2](https://github.com/elsif2) wants to merge 3 commits into [develop](https://github.com/certtools/intelmq) from [shadowserver-202207](https://github.com/certtools/intelmq/tree/shadowserver-202207)

sebix commented:
Cool, looks tidy now :)

aaronkaplan commented:
Agreement is that I merge it in, if things break, @elsif2 will fix it quickly. 
IMHO it is very important to have the new shadowserver code in 3.1. However, please note that we should really test this in a sample environment before tagging the release. Agreed, @sebix ?

Thanks @elsif2 for your enormous heavy lifting and improvements!
Can't over-state how valuable this is.
